File: SDK/java/src/main/java/com/reloop/verifio/HistoryService.java
package com.reloop.verifio;

import com.reloop.verifio.types.PaginatedData;
import com.reloop.verifio.types.PaginationOptions;
import com.reloop.verifio.types.VerificationResult;

public class HistoryService {
    private final VerifioClient client;

    public HistoryService(VerifioClient client) {
        this.client = client;
    }

    public PaginatedData<VerificationResult> list(PaginationOptions options) {
        int page = options != null && options.getPage() != null ? options.getPage() : 1;
        int limit = options != null && options.getLimit() != null ? options.getLimit() : 20;

        String path = "/history?page=" + page + "&limit=" + limit;

        return this.client.request("GET", path, null, VerificationResultPaginated.class);
    }

    public static class VerificationResultPaginated extends PaginatedData<VerificationResult> {}
}

File: SDK/.net/HistoryService.cs
using System.Net.Http;
using System.Threading.Tasks;
using Verifio.Models;

namespace Verifio
{
    public class HistoryService
    {
        private readonly VerifioClient _client;

        public HistoryService(VerifioClient client)
        {
            _client = client;
        }

        public Task<PaginatedData<VerificationResult>?> ListAsync(PaginationOptions? options = null)
        {
            var page = options?.Page ?? 1;
            var limit = options?.Limit ?? 20;

            return _client.RequestAsync<PaginatedData<VerificationResult>>(HttpMethod.Get, $"/history?page={page}&limit={limit}");
        }
    }
}
